### ISSUE TITLE
[DLB]2018-01-27 질문사항

### ISSUE THREAD

**Issue by SongChiYoung** [YBIGTA/data-science-2018#8]
[DLB 2주차](https://github.com/YBIGTA/DLB/blob/master/Neural%20Networks%20and%20Deep%20Learning/2.%20Neural%20Networks%20Basics.md)

**Comment by statkim7578** [comment]
![gd](https://user-images.githubusercontent.com/34841210/35448594-06ef3990-02fe-11e8-9223-a9fb4521a8f5.PNG)

Cost function을 최적화하는데 있어서 Gradient Descent말고도 뉴튼랩슨 방법을 쓰기도 하나요?
뉴튼랩슨방법은 비선형방정식의 근을 찾는 방법이긴 하지만,  f의 극소값 찾는문제도 결국 f'=0의 근을 찾는 문제로 바꿔풀수 있을것 같아서요. 

만약 뉴튼랩슨 방법으로도 최적화를 할 수 있는데 굳이 안하는거라면, Gradient Descent 방법이 뉴튼랩슨보다 수렴속도가 빠르기때문이라 생각해도 될까요?

**Comment by KangBK0120** [comment]
@statkim7578 
https://stats.stackexchange.com/questions/253632/why-is-newtons-method-not-widely-used-in-machine-learning

이글을 보면 실제로 로지스틱 회귀를 할때에는 뉴튼랩슨 방법을 사용하기는 하지만  필요한 연산이 Newton's method에서 더 많다고 하네요. 그렇기 때문에 Gradient descent를 사용한다고 합니다. 질문을 제대로 이해했는지 모르겠네요ㅜㅜ

**Comment by hyejukim-ybigta** [comment]
loss fucntion을 미분하는 법을 알았으니, cost function(J(w,b))도 미분할 수 있습니다..!  << loss function 미분은 전체 데이터에서 w,b 값을 찾는 것이고 cost function 미분하는 이유는 각각의 데이터에서 최적화된 w,b 값을 찾는 것인가요?

**Comment by ybigtaJH** [comment]
logistic regression에서 파라미터를 추정할 때, likelihood function을 최대화하는 파라미터의 값을 찾는maximum likelihood estimation(MLE)을 쓴다고 알고 있는데, 이 방법과 loss function은 어떤 관계가 있나요??

**Comment by godongyoung** [comment]
'w와 x를 내적(wT.x)하고 b를 더한 값은 0과 1사이의 값이 아닐 수 있습니다. sigmoid 함수를 사용하여 이 값을 0과 1사이의 값으로 만듭니다. sigmoid함수는 σ(x)라 쓰고, 식은 위의 사진에 나와 있습니다. '
에서 표현이 약간 혼동이 왓는데, LR을 한 뒤에 sigmoid를 씌워준다는 말이 아니라 sigmoid(logistic function)를 해주는것 까지가 LR인 것이지요?

**Comment by jungsooyun** [comment]
위에서 말한 뉴튼 랩슨 방식은 무엇인가요!! 아주 작은 일정한 Learning Rate로 근을 찾아나가는 방식인가요?
그러면 gradient descending이랑 다른 점은 learning rate 업데이트를 하는가의 여부인가요?

**Comment by Jooong** [comment]
> 그 길을 따라 (Learning rate - α)만큼 내려가는(w, b를 update) 것입니다.

이 부분이 이해가 안가는데요, 그림에서 α가 learning rate 아닌가요?!

**Comment by DabinJeong** [comment]
예전에 Gradient descent에서 Learning rate가 너무 크면 극소값으로의 수렴이 어렵다고 배웠었는데..이 때 적절한 learning rate는 어떻게 정하나요??

**Comment by godongyoung** [comment]
<img width="535" alt="ques" src="https://user-images.githubusercontent.com/31824102/35468150-48d89664-0310-11e8-951e-a1723304db5e.PNG">
수식에 오타가 난것같은데, '-'의 괄호가 전체를 감싸야하는것 인지요?
만약 맞다면 그게 종훈이를 혼란에 빠트린 이유같습니다

**Comment by Jooong** [comment]
@Rafaela94 실험해보면서 적정한 lr을 찾아야겠지만, optimizer에 따라서 대충 lr 초기값을 비슷하게 주기는 합니다. (sgd: 1e-2, adam: 1e-4 등...) 이후 training epoch에 따라서, 혹은 validation score를 살펴보면서 decay를 해주기도 합니다. 최근에는 [Cyclic learning rate schedule](https://techburst.io/improving-the-way-we-work-with-learning-rate-5e99554f163b)의 개념도 등장했습니다.

**Comment by projokim** [comment]
@Jooong 
제가 헷갈리게 썼네요...learning rate = a 란 의미로 쓴 것입니다...!

**Comment by projokim** [comment]
@hyejukim-ybigta  
loss는 데이터 각각의 에러를 나타내고, cost는 데이터 전체의 평균에러를 나타내므로 loss를 미분하여 평균내면 그게 cost의 미분값이 된다는 뜻입니다!

**Comment by projokim** [comment]
@godongyoung 
아...넵 맞습니다!!! 괄호가 하나 빠졌네요....전체를 감싸는 -입니다!

**Comment by projokim** [comment]
@ybigtaJH 
loss function을 정확히 말하면 negative log-likelyhood function입니다! 제 기억에..........likelihood function에 로그를 씌우고 마이너스를 붙인  것인것 같은데......미분하기 더 용이하다고 들었던 것 같은데.....더 알아보고 오겠습니다...

**Comment by Jooong** [comment]
https://github.com/YBIGTA/data-science-2018/issues/8#issuecomment-360856782
테스트
